core/memslot.c

```diff
@@ -364,8 +364,12 @@ static inline void memslot_delete(hax_memslot *dest)
 
 static inline void memslot_move(hax_memslot *dest, hax_memslot *src)
 {
+    hax_list_node entry;
+
     ramblock_deref(dest->block);
+    entry = dest->entry;
     memslot_init(dest, src);
+    dest->entry = entry;
```

Comment:
Many thanks for @leecher1337 shooting the root cause! For performance reasons, it is recommended to get rid of memslot_init() invocation to avoid assigning dest->entry 3 times within this function. I drafted the commit message for your reference. The following code should be equivalent for your reference. Thanks again.

Fix a bug of memslot_move()

Invoking memslot_init() in memslot_move() will result in losing the
linked list information of the original memory slot node. Reimplement
memslot_move() to resolve some exceptional issues during mapping memory
slots.

static inline void memslot_move(hax_memslot *dest, hax_memslot *src)
{
    ramblock_deref(dest->block);
    src->entry = dest->entry;
    *dest = *src;
    ramblock_ref(dest->block);
}